e2e_tests/tests/cluster/test_groups.py

```diff
@@ -0,0 +1,132 @@
+import json
+import subprocess
+from typing import Any, List
+
+import pytest
+
+from tests import config as conf
+
+from .test_users import get_random_string
+
+
+def det_cmd(cmd: List[str]) -> subprocess.CompletedProcess:
+    return subprocess.run(
+        ["det", "-m", conf.make_master_url()] + cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+
+
+def det_cmd_json(cmd: List[str]) -> Any:
+    res = det_cmd(cmd)
+    assert res.returncode == 0
+    return json.loads(res.stdout)
+
+
+def det_cmd_expect_error(cmd: List[str], expected: str) -> None:
+    res = det_cmd(cmd)
+    assert res.returncode != 0
+    assert expected in res.stderr.decode()
+
+
+@pytest.mark.e2e_cpu
+@pytest.mark.parametrize("add_users", [[], ["admin", "determined"]])
+def test_group_creation(add_users: List[str]) -> None:
+    group_name = get_random_string()
+    create_group_cmd = ["user-group", "create", group_name]
+    for add_user in add_users:
+        create_group_cmd += ["--add-user", add_user]
+
+    assert det_cmd(create_group_cmd).returncode == 0
+
+    # Can view through list.
+    group_list = det_cmd_json(["user-group", "list", "--json"])
+    assert sum([group["group"]["name"] == group_name for group in group_list["groups"]]) == 1
+
+    # Can view through list with userID filter.
+    for add_user in add_users:
+        group_list = det_cmd_json(
+            ["user-group", "list", "--json", "--groups-user-belongs-to", add_user]
+        )
+        assert sum([group["group"]["name"] == group_name for group in group_list["groups"]]) == 1
+
+    # Can describe properly.
+    group_desc = det_cmd_json(["user-group", "describe", group_name, "--json"])
+    assert group_desc["name"] == group_name
+    for add_user in add_users:
+        assert sum([u["username"] == add_user for u in group_desc["users"]]) == 1
+
+    # Can delete.
+    assert det_cmd(["user-group", "delete", group_name, "--yes"]).returncode == 0
```

Comment:
will that print the error output when it fails? if no then maybe

```
def det_cmd(cmd: List[str], **kwargs) -> subprocess.CompletedProcess:
    return subprocess.run(
        ["det", "-m", conf.make_master_url()] + cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs
    )
```

and here

```suggestion
    det_cmd(["user-group", "delete", group_name, "--yes"], check=True)
```